pkg/operator/migration_aws_status/controller.go

```diff
@@ -0,0 +1,132 @@
+	clusterConfigName      = "cluster-config-v1"
+	clusterConfigKey       = "install-config"
+)
+
+// MigrationAWSStatusController migrates the existing `infrastructure.config.openshift.io/v1` `cluster` objects
+// for AWS to include the `AWSPlatformStatus` based on the InstallConfig stored in the ConfigMap `kube-system/cluster-config-v1`.
+// BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1814332
+// The controller reads the configmap for the `install-config.yaml` and then creates a `AWSPlatformStatus` and updates the infrastructure object with these values.
+// Here are the values that are migrated by the controller:
+// - `.status.platformStatus.type` = `.status.platformType`
+// - `.status.platformStatus.aws.region` = AWS region from InstallConfig's `.platform.aws.region`
+//
+// The controller does no-op for non-AWS platforms.
+// It uses the `.status.platformType` from infrastructure object to identify the platform.
+type MigrationAWSStatusController struct {
+	infraClient     configv1client.InfrastructureInterface
+	infraLister     configv1listers.InfrastructureLister
+	configMapClient corev1client.ConfigMapsGetter
+}
+
+// NewController returns a MigrationAWSStatusController
+func NewController(operatorClient operatorv1helpers.OperatorClient,
+	infraClient configv1client.InfrastructuresGetter, infraLister configv1listers.InfrastructureLister, infraInformer cache.SharedIndexInformer,
+	configMapClient corev1client.ConfigMapsGetter,
+	kubeSystemInformer cache.SharedIndexInformer,
+	recorder events.Recorder) factory.Controller {
+	c := &MigrationAWSStatusController{
+		infraClient:     infraClient.Infrastructures(),
+		infraLister:     infraLister,
+		configMapClient: configMapClient,
+	}
+	return factory.New().
+		WithInformers(
+			operatorClient.Informer(),
+			infraInformer,
+			kubeSystemInformer,
+		).
+		WithSync(c.sync).
+		WithSyncDegradedOnError(operatorClient).
+		ResyncEvery(time.Minute).
+		ToController("MigrationAWSStatusController", recorder)
+}
+
+func (c MigrationAWSStatusController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	obji, err := c.infraLister.Get("cluster")
+	if errors.IsNotFound(err) {
+		syncCtx.Recorder().Warningf("MigrationAWSStatusController", "Required infrastructures.%s/cluster not found", configv1.GroupName)
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	currentInfra := obji.DeepCopy()
+
+	if currentInfra.Status.Platform != configv1.AWSPlatformType || // not aws
+		(currentInfra.Status.PlatformStatus != nil && currentInfra.Status.PlatformStatus.Type != configv1.AWSPlatformType) || // not aws
+		(currentInfra.Status.PlatformStatus != nil && currentInfra.Status.PlatformStatus.Type == configv1.AWSPlatformType &&
+			currentInfra.Status.PlatformStatus.AWS != nil && len(currentInfra.Status.PlatformStatus.AWS.Region) > 0) { // aws, but the region is already set. so no need to migrate
+		return nil // no action
+	}
+
+	cc, err := loadClusterConfig(ctx, c.configMapClient)
```

Comment:
if the cluster config isn't present, short-circuit please.  Our end goal should be removal.

Reply:
for this migration to run the cluster-config is required. so i don't think the short-circuit makes sense.

if the cluster-config is gone and this migration still needs to run, that's an error.
this migration controller is going to go away before we think about dropping the cluster-config